models/src/main/scala/coop/rchain/models/EqualsM.scala

```diff
@@ -98,6 +99,17 @@ object EqualM extends EqualMDerivation {
   implicit val MatchEqual            = gen[Match]
 
   implicit val ConnectiveEqual = gen[Connective]
+  implicit def SignedEqual[A: EqualM] = new EqualM[Signed[A]] {
+    override def equal[F[_]: Sync](self: Signed[A], other: Signed[A]): F[Boolean] =
+      EqualM[A]
+        .equal(self.data, other.data)
+        .map(
+          dataEquals =>
+            dataEquals &&
+              self.sigAlgorithm == other.sigAlgorithm &&
```

Comment:
I think it is easier to check signatures first (less expensive check) and check data after (more expensive).